apps/console/src/features/applications/pages/application-edit.tsx

```diff
@@ -150,7 +151,33 @@ const ApplicationEditPage: FunctionComponent<ApplicationEditPageInterface> = (
             && applicationTemplates instanceof Array
             && applicationTemplates.length > 0)) {
 
-            return;
+            /**
+             * What's this?
+             *
+             * When navigating to an application using the direct url i.e.,
+             * /t/foo/develop/applications/:id#tab=1 this component will be
+             * mounted. But you see this {@link applicationTemplates}?; it is
+             * loaded elsewhere. For some reason, requesting the state from
+             * {@link useSelector} always returns {@code undefined} when
+             * directly navigating or refreshing the page. Therefore; it hangs
+             * without doing anything. It will show a overlay loader but
+             * that's it. Nothing happens afterwards.
+             *
+             * So, as a workaround; if for some reason, the {@link useSelector}
+             * return no data, we will manually emit the event
+             * {@link ApplicationActionTypes.SET_APPLICATION_TEMPLATES}. So, doing
+             * that ensures we load the application templates again.
+             *
+             * Consider this as a **failsafe workaround**. We shouldn't rely
+             * on this. This may get removed in the future.
+             */
+            const timer = setTimeout(() => {
+                ApplicationTemplateManagementUtils
```

Comment:
If `applicationTemplates` is not there in the redux state. Simply `dispatch` an action to get the templates again. Do we need a timeout here? @yasinmiran

Reply:
Refer: https://github.com/wso2/identity-apps/blob/master/apps/console/src/features/applications/components/application-list.tsx#L182

Reply:
`getApplicationTemplates` contains the dispatch logic right?

And `timeout` was added to delay the calls if to prevent multiple calls to the same method. We have `applicationTemplates`, `application` as `deps` we could entirely ignore the timeout as it's not a costly call.

Reply:
Removed timeout wrapper in commit ae22ac7